kasa/discover.py

```diff
@@ -378,20 +380,30 @@ def _get_device_instance(
                 f"Unable to read response from device: {ip}: {ex}"
             ) from ex
 
-        if (
-            discovery_result.mgt_encrypt_schm.encrypt_type == "KLAP"
-            and discovery_result.mgt_encrypt_schm.lv is None
-        ):
+        if discovery_result.mgt_encrypt_schm.encrypt_type in ("KLAP", "AES"):
             type_ = discovery_result.device_type
             device_class = None
-            if type_.upper() == "IOT.SMARTPLUGSWITCH":
-                device_class = SmartPlug
+            supported_device_types: dict[str, Type[SmartDevice]] = {
+                "SMART.TAPOPLUG": TapoPlug,
+                "SMART.KASAPLUG": TapoPlug,
+                "IOT.SMARTPLUGSWITCH": SmartPlug,
```

Comment:
Just a note for future, so no need for changes now, but we may want to move the mapping about supported device types inside the implementation classes.

Reply:
Captured this question [here](https://github.com/python-kasa/python-kasa/discussions/559)